### ISSUE TITLE
Feature Request: Mouse click bindings for drawing on slides

### ISSUE THREAD

**Issue by drupol** [jahnf/Projecteur#213]
Hello,

I'm an avid user of [pdfpc](https://pdfpc.github.io/) and frequently use it for presentations, particularly with the Logitech Spotlight 2 as my presentation pointer.

One of `pdfpc`’s features I find very useful is the ability to "draw" on slides during a presentation. Currently, I'm looking to enhance this functionality by integrating it with the capabilities of the Logitech Spotlight and Projecteur.

I've noticed that Projecteur allows various bindings, such as "Toggle spotlight", "key sequences", "cycle presets", but it does not support a binding that simulates a mouse click, which is essential for initiating and ending the draw function via the pointer.

**Feature Request:**

I propose adding a new binding option—“Key click short|long”—to the list of customizable bindings. This would allow the Logitech Spotlight (or similar devices) to activate the draw function on slides with a short or long press, mimicking a mouse click.

This would also open the doors to some other kind of interactions, not only for drawing. 

**Use Case:**

This feature would streamline the process for presenters who rely on their pointers to interact with their slides, making it a more seamless and integrated experience.

I believe this enhancement would benefit many users who seek a more interactive presentation experience.

Thank you for considering this request, and I look forward to any possibility of its implementation.

**Comment by jahnf** [comment]
Yes this is not possible right now. So if I understand correctly what you are looking for is a mapping functionality that can map to 
* press 'mouse' button and hold...
* release pressed mouse button

Or maybe even better to have it as an option for the 'press and hold'..

How would you activate the drawing functionality in the mentioned pdf software? Would that be an additional shortcut?

In general I am unsure if you could 'draw' in a meaningful way :)

**Comment by drupol** [comment]
Thank you for your response.

As it stands, the functionality to draw on slides as you described is not available in the state of things. The ultimate goal is to replicate one of the interactive drawing capability similar to that of an Apple Pencil, which allows for temporary highlighting (with any colour) on slides that fades after a set period (configurable).

Just like you said, to achieve this, we would require the implementation of a "press and hold" binding or something similar.

> How would you activate the drawing functionality in the mentioned pdf software? Would that be an additional shortcut?

This could be a key sequence like: "if `<spotlight-button-A-is-hold>` then `<press> 3, <hold-left-button-mouse>`"
If this is not feasible, we could just press a key on the keyboard to enable that mode in `pdfpc`.

> In general I am unsure if you could 'draw' in a meaningful way :)

Even if the stability of the Logitech Spotlight 2 is impressive, I agree that these kinds of tools are not made for drawing. The idea of this feature is just to highlight things temporary on the screen.

**Comment by Blaimi** [comment]
I would also like a "drag" functionality :-) – independend of pdfpc ;-). I would also suggest a click+release+click&hold similar to touchpads when tap to click is enabled.
